state-sync/src/request_manager.rs

```diff
@@ -113,14 +114,20 @@ impl RequestManager {
         }
     }
 
-    pub fn enable_peer(&mut self, peer: PeerNetworkId, origin: ConnectionOrigin) {
+    pub fn enable_peer(
+        &mut self,
+        peer: PeerNetworkId,
+        origin: ConnectionOrigin,
+    ) -> Result<(), Error> {
         let is_upstream_peer = self.is_upstream_peer(&peer, origin);
         debug!(LogSchema::new(LogEntry::NewPeer)
             .peer(&peer)
             .is_upstream_peer(is_upstream_peer));
-
         if !is_upstream_peer {
-            return;
+            return Err(Error::PeerIsNotUpstream(
+                format!("{}", peer),
+                format!("{}", origin),
```

Comment:
You don't need to use format since it implements display:
`peer.to_string()`
`origin.to_string()`

How does this change the behavior, rather than just ignoring the peer?

Reply:
Thanks! Regarding behavior changes, it'll:
- Log the error (instead of just dropping it). I think this is useful.
- Not call "check_progress", which is acceptable, because check_progress is called through other means (e.g., the interval timer)